servers/servertcp.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+/**
+ * Copyright (c) 2017, Yaacov Zamir <kobi.zamir@gmail.com>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF  THIS SOFTWARE.
+ */
+ var util = require('util');
+ var events = require('events');
+ var EventEmitter = events.EventEmitter || events;
+ var net = require('net');
+
+ var HOST = '127.0.0.1';
+ var MODBUS_PORT = 502;
+
+/* Add bit operation functions to Buffer
+ */
+require('../utils/buffer_bit')();
+var crc16 = require('./../utils/crc16');
+
+/**
+ * Parse a modbusRTU buffer and return an answer buffer
+ */
+function parseModbusBuffer(buf, vector) {
+    var ansBuf = null;
+    var unitNumber = buf[0];
```

Comment:
Most of the time you use `unitID` in the existing code. I guess it makes sense to stay consistent in the naming of things.